File: src/main/java/me/ste/stevesseries/fancydrops/listener/PlayerListener.java
package me.ste.stevesseries.fancydrops.listener;

import me.ste.stevesseries.fancydrops.item.FancyItem;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerListener implements Listener {
    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        for(FancyItem item : FancyItem.ITEMS.values()) {
            if(item.getObservers().contains(event.getPlayer())) {
                item.removeObserver(event.getPlayer());
            }
        }
    }
}

File: src/main/kotlin/me/ste/stevesseries/fancydrops/listener/PlayerListener.kt
package me.ste.stevesseries.fancydrops.listener

import me.ste.stevesseries.fancydrops.item.FancyItem
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

object PlayerListener : Listener {
    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        for ((_, item) in FancyItem.ITEMS) {
            if (event.player in item.observers) {
                item.removeObserver(event.player)
            }
        }
    }
}